chat/chat-invite.md

```diff
@@ -0,0 +1,38 @@
+# Chat Invite
+
+## User Flow
+
+Once A has knowledge of the peer B's public key it can invite into a chat thread and with the following flow initiate a conversation:
+
+1. A sends an invite to B's public key and adds an opening message
+2. B receives the chat invite with an opening message
+3. B accepts the chat invite and the thread is created
+4. A receives notification the chat was accepted and created
+
+Now both A and B can exchange messages in the newly created chat thread
+
+## Invite Protocol
+
+A retrieves the public key associated with B's blockchain account, publicKey X.
+
+A generates a keyPair Y to encrypt the opening message with derived DH symKey I.
+
+A sends invite unecrypted to the invite topic.
+
+Invite topic is derived as the hash of the publicKey X.
+
+B decrypts opening message with the privateKey Y and deriving DH symKey I.
+
+B accepts the invite and generates a keyPair Z for chat thread.
+
+B sends response with publicKey Z on invite topic.
```

Comment:
the question is whether the response should be encrypted or not? 
on one hand it is "cheap" to do so 
on the other - the request is not, so response could be not encrypted for consistency

Reply:
Yeah I feel like it's a weird design decision to not encrypt a request and then encrypt the response. That being said it's also more privacy-preserving because you wouldn't be able acceptance from rejection